### PR TITLE
DomComponentWrapper - update options only if they are changed

### PR DESCRIPTION
**Pull request by novsstation**

### CODE REVIEW

File: js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts [hunk 1]
```diff
@@ -155,4 +155,22 @@ describe('get_updated_options', () => {
...
+  it('integration otions are ignored', () => {
+    expect(getUpdatedOptions(
+      { visible: true, integrationOptions: true },
+      { visible: false, integrationOptions: undefined },
+    )).toEqual([
+      { path: 'visible', value: false, previousValue: true },
+    ]);
+  });
+
+  it('integration otions in child props are ignored', () => {
```
Comment (MikeVitik) [inline js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts]: typo 'otions' -> 'options'

File: js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { getUpdatedOptions } from '../get_updated_options';
+import { getUpdatedOptions } from '../../../../common/utils/get_updated_options';
```
Comment (MikeVitik) [inline js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts]: Move this file to /ui/common/test

File: js/renovation/ui/common/dom_component_wrapper.tsx [hunk 3]
```diff
@@ -57,6 +58,9 @@ export class DomComponentWrapper extends JSXComponent<DomComponentWrapperProps,
   @Mutable()
   instance!: DomComponent | null;
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  @Mutable() prevProps!: any;
```
Comment (MikeVitik) [inline js/renovation/ui/common/dom_component_wrapper.tsx]: Why any but not Record<string, unknown>

File: js/renovation/ui/common/utils/get_updated_options.ts [hunk 4]
```diff
@@ -94,5 +94,6 @@ export function getUpdatedOptions(
   props: {},
   notDeepCopyArrays: string[] = defaultNotDeepCopyArrays,
 ): ResultItem[] {
-  return objectDiffsWithoutReactProps(oldProps, props, '', notDeepCopyArrays);
+  return objectDiffsWithoutReactProps(oldProps, props, '', notDeepCopyArrays)
+    .filter((item) => !item.path.includes('integrationOptions'));
```
Comment (MikeVitik) [inline js/renovation/ui/common/utils/get_updated_options.ts]: May be add 'integrationOptions' into reactProps?

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: js/renovation/ui/common/dom_component_wrapper.tsx (hunk 3 above)
Comment (novsstation) [reply js/renovation/ui/common/dom_component_wrapper.tsx]: Yep, fixed

File: js/renovation/ui/common/utils/get_updated_options.ts (hunk 4 above)
Comment (novsstation) [reply js/renovation/ui/common/utils/get_updated_options.ts]: reactProps checking is not recursive, so it's not used in the checking of children props.
But idea is great, so I skip integration props during comparing of the old and new props (not after)

File: js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts (hunk 2 above)
Comment (novsstation) [reply js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts]: Yep, fixed

File: js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts (hunk 1 above)
Comment (novsstation) [reply js/renovation/ui/grids/data_grid/utils/__tests__/get_updated_options.test.ts]: Yep, fixed

**Review by MikeVitik — APPROVED** [review]

**Pull request merged**